File: src/task2_pkg/src/complex.cpp
#include <ros/ros.h>
#include "task2_pkg/complex_num.h"
#include <random>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "publisher_node");
  ros::NodeHandle nh;
  ros::Publisher pub = nh.advertise<task2_pkg::complex_num>("sensor_topic", 10);
  ros::Rate rate(0.5);

  while (ros::ok())
  {
    task2_pkg::complex_num msg;
    msg.real = std::rand() * 2 - 1;
    msg.imaginary = std::rand() * 2 - 1;
    pub.publish(msg);
    rate.sleep();
    std::cout << "Complex number: " << msg.real << " + " << msg.imaginary << "i" << std::endl;
  }

  return 0;
}

File: src/task2_pkg/scripts/complex.py
#!/usr/bin/env python3
import rospy
from task2_pkg.msg import complex_num
import random

pub = rospy.Publisher('sensor_topic', complex_num, queue_size=10)
rospy.init_node('publisher_node', anonymous=True)
rate = rospy.Rate(0.5) # 1hz

while not rospy.is_shutdown():
    num = complex_num()
    num.real = random.random()
    num.imaginary= imaginary = random.random()
    rospy.loginfo("Complex_number:")
    rospy.loginfo(num)
    pub.publish(num)
    rate.sleep()